pkg/porter/publish.go

```diff
@@ -57,6 +57,21 @@ func (o *PublishOptions) Validate(cxt *portercontext.Context) error {
 		return o.validateReference()
 	}
 
+	if o.Tag != "" {
+		return o.validateTag()
+	}
+
+	return nil
+}
+
+// validateTag checks to make sure the supplied tag is of the expected form.
+// A previous iteration of this flag was used to designate an entire bundle
+// reference.  If we detect this attempted use, we return an error and
+// explanation
+func (o *PublishOptions) validateTag() error {
```

Comment:
Thanks for adding this! It will really help avoid a bunch of very unhappy users.